src-ghc-9.4/GHC/TypeLits/Extra/Solver.hs

```diff
@@ -181,7 +181,12 @@ simplifyExtra defs eqs = tcPluginTrace "simplifyExtra" (ppr eqs) >> simples [] [
           | otherwise     -> return  (Impossible eq)
         (p, Max x y)
           | b && (p == x || p == y) -> simples (((,) <$> evMagic ct <*> pure ct):evs) news eqs'
-
+        -- transform:  Mod n p <= q
```

Comment:
Normally `Mod x 0 <= q` would not reduce, as `Mod x 0` has no solutions. So I think we should also emit a `1 <= p` constraint.